search/serializers.py

```diff
@@ -339,3 +395,29 @@ def serialize_comment_for_bulk(comment_obj):
     except NotFound:
         log.exception("Reddit comment not found: %s", comment_obj.id)
         raise
+
+
+def serialize_bulk_courses(ids):
+    """
+    Serialize courses for bulk indexing
+
+    Args:
+        ids(list of int): List of course id's
+    """
+    for course in Course.objects.filter(id__in=ids).prefetch_related(
+        "topics", "instructors", "prices"
```

Comment:
I'm not 100% certain about this, but I think the only prefetch that provides benefit here is `instructors` because you walk `instructors.all()` in the serializer. `get_topics` and `get_prices` actually generate _new_ queries, so including the other fields in this prefetch counter-intuitively adds overhead.

There are probably two options here:

- Don't worry about trying to prefetch those other fields and we'll worry about it if/when it becomes a bad pain point
- Use [`Prefetch`](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#prefetch-objects) to set attrs as for `topics` and `prices`. An example would be:

```python
from django.db.models import Prefetch

topic_names = CourseTopic.objects.values_list("name", flat=True)
prefetch_topic_names = Prefetch("topics", queryset=topic_names, to_attr="topic_names")
Course.objects.filter(id__in=ids).prefetch_related(prefetch_topic_names)
```

And then it can just be a simple `ListField` on the serializer.

The major downside to this approach is it very tightly couples the serializer implementation to the queryset it is passed.

I'd probably go for the former option, but I just wanted to play out the alternative.

Reply:
Went with the former